### ISSUE TITLE
scalar_minimize does not copy attributes

### ISSUE THREAD

**Issue by woutergins** [lmfit/lmfit-py#285]
When using the scalar_minimize method, the message and success boolean from scipy's minimize are not copied to the MinimizerResult. This should be done in (`minimizer.py`, lines 484-486)
```
for attr in dir(ret):
    if not attr.startswith('_'):
        setattr(result, attr, getattr(ret, attr))
```
but it does not appear to work. Hardcoding in the setting of these attributes does work, so
```
result.success = ret.success
result.message = ret.message
```

**Comment by newville** [comment]
@woutergins  Thanks... It looks like scipy.optimize.OptimizeResult became less object-like and more dictionary-like, perhaps in a very recent version.  The code should be changed to look at the "items()" of "ret".  Probably we need to add some sort of test for which version is being used?

**Comment by newville** [comment]
should be addressed with a8cbeb5978de84332bb92dd5f446e494c023f272
